### DISCUSSION TITLE
Sharing dashboards externally

### DISCUSSION THREAD

**Discussion by TomOgs** [telekom-security/tpotce#877] [Q&A]
Hey all. 

I'm attempting to share dashboards from Kiabana on an external site. 

Due to security issues with cookies, I was attempting to change the "xpack.security.sameSiteCookies" in the kibana.yml file to allow the data to be shared to an external site. 

Being quite new to this and Docker setups, I couldn't locate the kibana.yml file to make the changes. 

Sorry for the noob question, but could someone point me towards where the file is located on a standard installation. 

Much appreciated.

**Comment by Ganesh-403** [comment]
@TomOgs 
The `kibana.yml` file isn't typically edited directly on the host in a standard T-Pot installation because **Kibana runs inside a Docker container** and its configuration is managed by the container/compose setup rather than a native package installation. :contentReference[oaicite:0]{index=0}

If your goal is to allow Kibana dashboards to be embedded on an external site by changing:

```yaml
xpack.security.sameSiteCookies: None
```

you'll also need to set:

```yaml
xpack.security.secureCookies: true
```

Elastic documents that `SameSite=None` requires secure cookies (HTTPS). Otherwise, modern browsers will reject the session cookie and embedding won't work. :contentReference[oaicite:1]{index=1}

For a standard T-Pot installation, I'd recommend **not editing files inside the running container**, as those changes will be lost when the container is recreated.

Instead:

1. Stop T-Pot:

   ```bash
   systemctl stop tpot
   ```

2. Locate the Kibana service definition in the T-Pot Docker Compose configuration (`/opt/tpot/etc/tpot.yml` on older releases, or `~/tpotce/docker-compose.yml` on newer releases). :contentReference[oaicite:2]{index=2}

3. Mount a custom `kibana.yml` into the Kibana container (or add the equivalent environment/configuration if supported by your T-Pot version).

4. Restart T-Pot:

   ```bash
   systemctl start tpot
   ```

One question:

- Which **T-Pot version** are you using (20.06.x, 22.04.x, or a newer `tpotce` release)?

The configuration layout changed significantly between the older `/opt/tpot` releases and the current `~/tpotce` Docker Compose deployment, so the exact location to customize Kibana depends on your version.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
